### PR TITLE
Electron

### LINKED ISSUE #27: Inconsistency with Electron
Electron `env` has require, while Monaco uses require as ADM loaded marker.

Sample: https://github.com/Microsoft/monaco-editor-samples/blob/master/sample-electron/index.html
Some more discussions: https://github.com/Microsoft/monaco-editor/issues/90

### PR DESCRIPTION
**Pull request by expobrain**
Fixed issue with `react-monaco-editor` in Electron and added an example too.

Fixes #27

### CODE REVIEW

**Comment by josiahruddell** [comment]
Found this PR from #27. Doing some testing, the electron conditional statements ignore the amdRequire config. In my scenario we run a webpack dev server in dev mode so the amd requires needs to go through `http:` vs `file:`. Maybe a consideration before merging.

**Comment by expobrain** [comment]
Thank you @josiahruddell for your feedback! 

I'll modify the example to run with a local webpack dev server as well and fix the issue

**Comment by agdolla** [comment]
sorry but it does not work for me
I use create-react-app (typescript variation from microsoft) and it does not load source from outside of src
so this won't work
```
const monacoPath = path.join(context.__dirname, '../node_modules/monaco-editor/min');
```
I do not want to eject

when I use this in browser environment I could make it to work by copying vs folder under public
so this PR could also be able to to so in Electron environment

**Comment by solemnify** [comment]
@josiahruddell @expobrain 

Any update on this by any chance? Blocked by this... if there's any example for referencing monaco in electron with http for dev mode, that would be amazing....

Thanks for your time.

**Comment by agdolla** [comment]
@solemnify  I also spent last two days with this problem
unfortunatelly loader.js itself handles electron differently

File: examples/electron/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# React+Monaco+Electron example
+
+## Run
```
Comment (domoritz) [inline examples/electron/README.md]: Maybe add a simple option: "npm run start" as well.

File: examples/electron/app.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import { render } from 'react-dom'
+
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable import/no-unresolved */
+/* eslint-disable import/extensions */
```
Comment (domoritz) [inline examples/electron/app.js]: why are all these necessary? Can you just disable all linting for this one line?

File: examples/electron/index.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+// Keep a global reference of the window object, if you don't, the window will
+// be closed automatically when the JavaScript object is garbage collected.
+let win;
+
+const createWindow = () => {
+  // Create the browser window.
+  win = new BrowserWindow();
+
+  // and load the index.html of the app.
```
Comment (domoritz) [inline examples/electron/index.js]: Load the index.html...

File: examples/electron/README.md (hunk 1 above)
Comment (expobrain) [reply examples/electron/README.md]: fixed in https://github.com/superRaytin/react-monaco-editor/pull/50/commits/6ce01b5cb8f4151deb8689c43e2e429c386fbe0d

**Pull request merged**